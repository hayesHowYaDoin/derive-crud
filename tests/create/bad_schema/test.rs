use derive_crud::Create;

#[derive(Create)]
#[crud_table("test_table")]
struct Test {
    #[crud_id]
    id: i64,
    name: String,
    location: String,
}

#[tokio::main]
async fn main() {
    Test::create("Debbie".to_string(), "New York".to_string())
        .await
        .expect("Failed to create entry");
}
