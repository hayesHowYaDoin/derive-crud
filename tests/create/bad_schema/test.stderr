error: error returned from database: (code: 1) table test_table has no column named location
 --> tests/create/bad_schema/test.rs:3:10
  |
3 | #[derive(Create)]
  |          ^^^^^^
  |
  = note: this error originates in the macro `$crate::sqlx_macros::expand_query` which comes from the expansion of the derive macro `Create` (in Nightly builds, run with -Z macro-backtrace for more info)
